### ISSUE TITLE
InverseLaplaceTransform not defined

### ISSUE THREAD

**Issue by zamlz** [sympy/sympy#26999]
I am attempting to lambdaify a inverse laplace transform. Is this not possible?

```
File <lambdifygenerated-1>:2, in _lambdifygenerated(time, white_noise)
      1 def _lambdifygenerated(time, white_noise):
----> 2     return 120.0*time*select([less(time, 0),equal(time, 0),True], [0,1/2,1], default=nan) - 2.0*InverseLaplaceTransform(300*pi*LaplaceTransform(floor(60*time + 0.5), time, laplace_var_s)/(laplace_var_s + 300*pi), laplace_var_s, time, None_1508566) - 0.4*select([less(time, 0),equal(time, 0),True], [0,1/2,1], default=nan)/pi + 0.4*exp(-300*pi*time)*select([less(time, 0),equal(time, 0),True], [0,1/2,1], default=nan)/pi

NameError: name 'InverseLaplaceTransform' is not defined
```

**Comment by sylee957** [comment]
It is difficult or not possible to implement the code generation for `LaplaceTransform` or `InverseLaplaceTransform`.
Such functions may not be constructive in Python or NumPy.

However, I think that `InverseLaplaceTransform` appears in failed attempt to solve integrals in SymPy,
and that indicates either if there is no closed form solution of integrals 
(and we can't help through this, because it proves mathematically that we can't write computer programs for that as well)
or if there is solution of integral, but SymPy just fails computing the answer.
And if you can share the original expression before lambdify, we can check the problem in more detail.

**Comment by zamlz** [comment]
Ahhh damn, that's what I was worried about. After I posted the issue last night I was playing with simpler functions and I was able to get it to solve.

I can post the equation I used later tonight but ultimately it was a sawtooth wave. I forget how exactly I defined it but it was with some functions like floor (I think). I redefined my sawtooth as a Fourier series of sine waves and that worked when the number of waves I was summing was like 5, but once I increased the resolution, things got slow. I am applying a low pass filter while in the Laplace domain and then running the inverse Laplace transformation and I haven't logged it, but this whole process just were waiting indefinitely. Ultimately killed it.

But it looks like to me that it was fundamentally because I was using functions that we can't ultimately apply Laplace transforms for.

Thanks for confirming what I feared lol.

-------- Original Message --------

**Comment by zamlz** [comment]
@sylee957 Okay, so I've been playing with different things and here is what I've learned. Please correct me if I am wrong. But it should be possible take integrals of piecewise functions. Therefore it should be possible to take laplace transformations of piecewise functions?

I've defined the following piecewise function and I want to take the laplace transformation of it.

```python
import sympy as sp

t = sp.symbols('t')
s = sp.symbols('s')

def square(f):
    return sp.Piecewise(
        (1, sp.sin(2*sp.pi*f*t) >= 0),
        (-1, sp.sin(2*sp.pi*f*t) < 0),
    )

lp = sp.laplace_transform(square(120), t, s, noconds=True)

# Out[16]: LaplaceTransform(Piecewise((1, sin(240*pi*t) >= 0), (-1, True)), t, s)

signal = sp.inverse_laplace_transform(lp, s, t)

# Out[20]: InverseLaplaceTransform(LaplaceTransform(Piecewise((1, sin(240*pi*t) >= 0), (-1, True)), t, s), s, t, _None)

```

My understanding is that if it has `LaplaceTransform` or `InverseLaplaceTransform` is that it was unable to solve it? Is that correct?

**Comment by hanspi42** [comment]
> My understanding is that if it has `LaplaceTransform` or `InverseLaplaceTransform` is that it was unable to solve it? Is that correct?

This is correct: I have not implemented this yet because I don't know a way that lets a computer parse such things in general.

If you want a solution: For f=1, your square function that I call $g(t)$ flips between 1 and -1 with a frequency of $f$. On page 141 of https://github.com/sympy/sympy-development-docs/blob/main/integrals/laplace/tables_laplace_transforms__bateman54.pdf, you have (1) with a function $f(t)$ that flips between 0 and 1 with repetition time $2b$, so $b=\frac1{2f}$.

Furthermore, $g(t)=1-2f(t)$, so the Laplace transform is

$\displaystyle \frac1s - \frac{2}{s\left(e^{\frac{s}{2f}}+1\right)}$

which can again be simplified.
